//Define the used version of proto
syntax = "proto3";

package aqpredictions;


message Empty {

}

message AQTrainRequest{
  string station = 1;
  string pollutant = 2;
  int32 threshold = 3;
  bool use_temporal = 4;
  bool use_delta = 5;
  bool use_only_pm = 6;
  bool use_weather = 7;
  bool use_forecast = 8;
  bool use_traffic = 9;
}

message AQTrainResponse{
  float test_recall = 1;
  float test_precision = 2;
  float test_roc = 3;
}

//Define a message to hold the features input by the client
message AQPredictRequest {
  float e6_tiller_pm25 = 1 ;
  float e6_tiller_pm10 = 2 ;
  float elgeseter_pm25 = 3 ;
  float elgeseter_pm10 = 4 ;
  float torvet_pm25 = 5 ;
  float torvet_pm10 = 6 ;
}
//Define a message to hold the response output
message AQPredictResponse {
  string predicted_aq_level = 1 ;
}


//Define the service
// TODO: change service name for something more concrete? (e.g., Predict) to distuinguish from other modules in the pipeline
service AI4IoT {
  rpc train_classifier(AQTrainRequest) returns (AQTrainResponse);
  rpc predict_aq_level(AQPredictRequest) returns (AQPredictResponse);
}